RockLib.Configuration.ObjectFactory/ConstructorOrderInfo.cs

```diff
@@ -56,6 +58,11 @@ public int CompareTo(ConstructorOrderInfo? other)
          if (TotalParameters < other.TotalParameters) return 1;
          if (MatchedNamedParameters > other.MatchedNamedParameters) return -1;
          if (MatchedNamedParameters < other.MatchedNamedParameters) return 1;
+         if (ParameterTypes is not null && other.ParameterTypes is not null)
+         {
+            if (ParameterTypes.Except(other.ParameterTypes).Any()) return -1;
+            if (other.ParameterTypes.Except(ParameterTypes).Any()) return 1;
```

Comment:
I'm wondering if there's a way to do this such that you don't have to do 4 passes over the enumerables. Offhand I don't have a suggestion, but....maybe just get the parameters as a list or array and then compare each one? Also, what happens if the parameter counts are different? Do we return 0 in that case, or 1/-1? And what's the right thing to do in that scenario?

Reply:
Updated it to a list.